packages/core/lib/debug/interpreter.js

```diff
@@ -737,6 +728,35 @@ class DebugInterpreter {
       this.lastCommand = cmd;
     }
   }
+
+  // update the printouts according to user inputs
+  // called by case v for section printouts and case p for location printouts
+  updatePrintouts(userArgs, selections, printOuts) {
```

Comment:
One final nitpick -- you should probably highlight in the comment here that this function is not pure, and that it *both* alters `printOuts` *and* returns a value.

I think your comment here does a good job of highlighting that this function is not pure and that it alters `printOuts`; but it, together with the name, potentially misleads the reader into thinking that that's all it does, and that it doesn't return a value.  I think you should explain the return value here as well.

Reply:
add comment:
```
  // update the printouts according to user inputs
  // called by case v for section printouts and case p for location printouts
  //
  // NOTE: THIS FUNCTION IS NOT PURE.
  //       The input printOuts is altered according to the values of other inputs: userArgs and selections.
  //       The function returns an object, tempPrintouts, that contains the selected printouts.
  updatePrintouts(userArgs, selections, printOuts) {
    let tempPrintouts = new Set();
```

Reply:
I think all caps is overkill, but it does get the message across! :P